iocage_lib/ioc_json.py

```diff
@@ -303,6 +322,53 @@ def zfs_get_snapshot(self, snap_id):
         # Snap_id expected value - vol/iocage/jails/jail1@snaptest
         return IOCSnapshot(snap_id)
 
+    def zfs_destroy_dataset(self, identifier, recursive=False, force=False):
+        cmd = ['zfs', 'destroy']
+
+        if recursive:
+            cmd += ['-r']
+
+        if force:
+            cmd += ['-Rf']
+
+        try:
+            su.run(
+                cmd + [identifier], check=True, stdout=su.PIPE, stderr=su.PIPE
+            )
+        except su.CalledProcessError as e:
+            if force:
+                return
+
+            iocage_lib.ioc_common.logit(
+                {
+                    'level': 'EXCEPTION',
+                    'message': f'Destroying {identifier} failed!\n'
+                               f'Reason: {e.stderr.decode()}'
+                },
+                _callback=self.callback,
+                exception=ioc_exceptions.CommandFailed
+            )
+
+    def zfs_get_dataset_and_dependents(self, identifier):
+        try:
+            datasets = list(su.run(
+                ['zfs', 'list', '-rHo', 'name', identifier],
+                check=True, stdout=su.PIPE, stderr=su.PIPE
+            ).stdout.decode().split())
+        except su.CalledProcessError as e:
+            iocage_lib.ioc_common.logit(
+                {
+                    'level': 'EXCEPTION',
+                    'message': 'Getting dataset and dependents for '
+                               f'{identifier} failed!\n'
+                               f'Reason: {e.stderr.decode()}'
+                },
+                _callback=self.callback,
```

Comment:
Would it be possible that this raises an undefined error if somewhere we use this function and callback isn't defined ?

Reply:
So if we use IOCZFS directly, it would not exist then